dokan_fuse/src/dokanfuse.cpp

```diff
@@ -433,6 +433,87 @@ static NTSTATUS DOKAN_CALLBACK FuseUnmounted(PDOKAN_FILE_INFO DokanFileInfo) {
   return errno_to_ntstatus_error(impl->unmounted(DokanFileInfo));
 }
 
+static DWORD AddSeSecurityNamePrivilege() {
+    LUID luid;
+    if (!LookupPrivilegeValue(0, SE_SECURITY_NAME, &luid)) {
+        DWORD err = GetLastError();
+        if (err != ERROR_SUCCESS) {
+            return err;
+        }
+    }
+
+    HANDLE token = 0;
+    if (!OpenProcessToken(GetCurrentProcess(),
+        TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &token)) {
+        return GetLastError();
+    }
+
+    ON_BLOCK_EXIT([token] {
+        CloseHandle(token);
+    });
+
+    LUID_AND_ATTRIBUTES attr;
+    attr.Attributes = SE_PRIVILEGE_ENABLED;
+    attr.Luid = luid;
+
+    TOKEN_PRIVILEGES priv;
+    priv.PrivilegeCount = 1;
+    priv.Privileges[0] = attr;
+
+    AdjustTokenPrivileges(token, FALSE, &priv, NULL, NULL, NULL);
+    DWORD err = GetLastError();
+    if (err != ERROR_SUCCESS) {
+        return err;
+    }
+
+    return ERROR_SUCCESS;
+}
+
+static NTSTATUS DOKAN_CALLBACK FuseGetFileSecurity(
+    LPCWSTR FileName, PSECURITY_INFORMATION SecurityInformation,
+    PSECURITY_DESCRIPTOR SecurityDescriptor, ULONG BufferLength,
+    PULONG LengthNeeded, PDOKAN_FILE_INFO DokanFileInfo) {
+  impl_fuse_context *impl = the_impl;
+  if (impl->debug())
+    FWPRINTF(stderr, L"GetFileSecurity %x\n", *SecurityInformation);
+
+  bool needASS = ((*SecurityInformation & SACL_SECURITY_INFORMATION) ||
+      (*SecurityInformation & BACKUP_SECURITY_INFORMATION));
+
+  if (needASS) {
+      DWORD result = AddSeSecurityNamePrivilege();
```

Comment:
@matepek does this should not be made in fuse_main_real ?
Like that it would only be made one time.

Also if AddSeSecurityNamePrivilege fail, it is better to ignore it. FuseGetFileSecurity could still work in different case.